### DISCUSSION TITLE
Slow tasks start

### DISCUSSION THREAD

**Discussion by itsai220** [concourse/concourse#6458] [Help & Support]
Hi guys,

i'm running concourse on GKE and deployed concourse via helm chart.
I've recently did a big upgrade jump from 5.8.0 to 6.4.1, and i noticed that my tasks now takes sometimes 6 minutes to start.

My containerPlacementStrategy is limit-active-tasks and currently i have limitActiveTasks set to 50. Are there any other config values i need to adjust to make task containers run faster ?

Thanks!

**Comment by taylorsilva** [comment]
Are your task containers `privileged` by any chance? Having privileged tasks slows down initialization because concourse has to `chown` all volumes for the task, which can take a while if your volumes have many files.

Another common reason is volume streaming. Did anything change in your topology of your Concourse? All volume streaming takes the path of `worker -> web -> worker` if the task and required volume(s) are on different workers. You could try changing to volume-locality and see if that brings down initialization time for your tasks.

**Reply by itsai220** [reply]
yes we do have privileged containers, i'm not sure if all impact ones have that i'l need to verify, however we have not changed our pipeline config only concousre version, was there a logic change at some point ? 

Yes i'm actually looking to switch to random, i want worker loads to spread more evenly and not eventually gravitate to certain workers.

**Reply by itsai220** [reply]
verified, most are not privileged containers that i see delays.

**Reply by itsai220** [reply]
switching container placement to random did not seem to help at all

**Comment by StevenArmstrong** [comment]
Check your db disk io is being maxed out and you have enough iops. This
caused us similar problems on 6.x on aws. We were getting throttled at busy
periods on the cloud provider and lidar algorithm continues and will
schedule all tasks but with terrible start times. Fixing that made
everything run fast. Your db cpu will be low when it happens so it was
harder to find.

**Reply by taylorsilva** [reply]
ohhh this is really good and makes sense. The big thing with 6.x is that the algorithm changed. Most of the algorithm logic used to happen in-memory on the web nodes. Starting in 6.x most of this work was off-loaded to the db.

**Reply by itsai220** [reply]
Thanks for the tips guys,

I've compared the iops on the db disk from before and after upgrade and it seems there is a significant drop in write ops
before upgrade
normal write iops is 100/s max 130/s

after upgrade
normal write iops is 40/s and highest point i see it reached was 100/s 

I'll also keep monitoring the quota limits , however with the metrics i don't think it's the hardware issue. I think this might be configuration issue. I'm assuming the values to tweak are lidarScannerInterval and lidarCheckerInterval, i believe these are new values (at least i don't remember seening these before in 5.x).
are there values that you guys recommend to start with ?
